### ISSUE TITLE
DDSM LJPEG optical density calibration

### ISSUE THREAD

**Issue by SteveBetter** [riblidezso/frcnn_cad#3]
Dear Dezso,
I want to train a neural network model on the DDSM dataset. When I do optical density calibration, I think I get wrong results. I posted my python function which should be obvious to you and two samples( C_0029_1.LEFT_CC.jpeg, C_0217_1.RIGHT_MLO.jpeg) below.

Could you spare some time on this?

Thanks very much!

> We have converted the lossless jpeg images to png format, mapped the pixel values to optical density using calibration functions from the DDSM website, and rescaled the pixel values to the 0–255 range.

`def optical_density_adjustment(image, case_dict):

    if case_dict["source_label"] == "A" and case_dict["machine"] == "DBA":
        image = ( np.log10(image) - 4.80662 ) / (-1.07553)
    elif case_dict["source_label"] == "A" and case_dict["machine"] == "HOWTEK":
         image = 3.789 - 0.00094568 * image
    elif case_dict["source_label"] == "B" or case_dict["source_label"] == "C" and case_dict["machine"] == "LUMISYS":
         image = 3.6 - (image - 495) / 1000
    elif case_dict["source_label"] == "D" and case_dict["machine"] == "HOWTEK":
         image = 3.96604095240593 + (-0.00099055807612) * image
    else:
         print("ERROR: undefined machine type")
         exit()
    image = cv2.normalize(image, None, 0, 255, cv2.NORM_MINMAX)
    image = np.uint8(image)
    return image
`
![c_0029_1 left_cc](https://user-images.githubusercontent.com/22906431/44375097-7d09cc80-a524-11e8-8797-ad71b0e643e1.jpeg)
![c_0217_1 right_mlo](https://user-images.githubusercontent.com/22906431/44375111-8b57e880-a524-11e8-9701-61a0e48b3c11.jpeg)

**Comment by SteveBetter** [comment]
I found something. After the optical density adjustment, 
`plt.imshow(image_n, "Greys")`
would be ok to show the above "abnormal" images.

With no optical density adjustment, 
`plt.imshow(image_n, "Greys_r")` 
should be used. This will show the same with ordinary Image Viewer Applications.

I will dig out what should be input to the neural networks.

**Comment by llllkihn** [comment]
Dear @SteveBetter
I am also trying to retrain network on the DDSM dataset.
DDSM images seem to have low contrast after optical density calibration.
Could you please remember whether the results of training the network using DDSM data with such transformation were comparable to the ones in the article?
Would be grateful for any info about this.
